src/MongoDB.Driver.Core/Core/Misc/ServerVersion.cs

```diff
@@ -0,0 +1,463 @@
+            {
+                releaseCandidate = releaseCandidateParsed;
+            }
+            int? internalBuild = null;
+            if (internalBuildEntry != null && int.TryParse(internalBuildEntry, out int internalBuildParsed))
+            {
+                internalBuild = internalBuildParsed;
+            }
+
+            result = new ServerVersion(major, minor, patch, releaseCandidate, internalBuild, commitHash);
+            return true;
+        }
+
+        // public operators
+        /// <summary>
+        /// Casts a SemanticVersion to a ServerVersion.
+        /// </summary>
+        /// <param name="semanticVersion">The SemanticVersion.</param>
+        /// <returns>A ServerVersion.</returns>
+        /// <exception cref="FormatException">semanticVersion</exception>
+        public static explicit operator ServerVersion(SemanticVersion semanticVersion)
+        {
+            return Parse(semanticVersion.ToString());
+        }
+
+        /// <summary>
+        /// Determines whether two specified server versions have the same value.
+        /// </summary>
+        /// <param name="a">The first server version to compare, or null.</param>
+        /// <param name="b">The second server version to compare, or null.</param>
+        /// <returns>
+        /// True if the value of a is the same as the value of b; otherwise false.
+        /// </returns>
+        public static bool operator ==(ServerVersion a, ServerVersion b)
+        {
+            if (ReferenceEquals(a, null))
+            {
+                return ReferenceEquals(b, null);
+            }
+
+            return a.CompareTo(b) == 0;
+        }
+
+        /// <summary>
+        /// Determines whether two specified server versions have different values.
+        /// </summary>
+        /// <param name="a">The first server version to compare, or null.</param>
+        /// <param name="b">The second server version to compare, or null.</param>
+        /// <returns>
+        /// True if the value of a is different from the value of b; otherwise false.
+        /// </returns>
+        public static bool operator !=(ServerVersion a, ServerVersion b)
+        {
+            if (ReferenceEquals(a, null))
+            {
+                return !ReferenceEquals(b, null);
+            }
+
+            return a.CompareTo(b) != 0;
+        }
+
+        /// <summary>
+        /// Determines whether the first specified ServerVersion is greater than the second specified ServerVersion.
+        /// </summary>
+        /// <param name="a">The first server version to compare, or null.</param>
+        /// <param name="b">The second server version to compare, or null.</param>
+        /// <returns>
+        /// True if the value of a is greater than b; otherwise false.
+        /// </returns>
+        public static bool operator >(ServerVersion a, ServerVersion b)
+        {
+            if (ReferenceEquals(a, null))
```

Comment:
It's confusing.. what's the purpose of using such construction?

Reply:
Checking for null without triggering == overload operator.